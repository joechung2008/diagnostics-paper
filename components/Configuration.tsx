import { useMemo } from "react";
import { Text, View } from "react-native";
import { useConfigurationStyles } from "../hooks/useConfigurationStyles";
import type { ConfigurationProps, KeyValuePair } from "../lib/types";

const Configuration = ({ config }: ConfigurationProps) => {
  const dynamicStyles = useConfigurationStyles();

  const items = useMemo(
    () =>
      Object.entries(config).reduce<KeyValuePair<string>[]>(
        (previous, [key, value]) => [...previous, { key, value }],
        []
      ),
    [config]
  );

  return (
    <View style={dynamicStyles.container}>
      <Text style={dynamicStyles.title}>Configuration</Text>
      <View style={dynamicStyles.table}>
        <View style={dynamicStyles.header}>
          <Text style={dynamicStyles.headerText}>Key</Text>
          <Text style={dynamicStyles.headerText}>Value</Text>
        </View>
        {items.map((item) => (
          <View key={item.key} style={dynamicStyles.row}>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.key}
            </Text>
            <Text style={[dynamicStyles.cell, dynamicStyles.cellText]}>
              {item.value}
            </Text>
          </View>
        ))}
      </View>
    </View>
  );
};

export default Configuration;
