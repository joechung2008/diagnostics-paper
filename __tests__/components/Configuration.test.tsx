import React from "react";
import { PaperProvider, MD3LightTheme } from "react-native-paper";
import { render } from "@testing-library/react-native";
import Configuration from "../../components/Configuration";

describe("Configuration", () => {
  it("renders correctly with config data", () => {
    const mockConfig = {
      key1: "value1",
      key2: "value2",
      key3: "value3",
    };

    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Configuration config={mockConfig} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly with empty config", () => {
    const { toJSON } = render(
      <PaperProvider theme={MD3LightTheme}>
        <Configuration config={{}} />
      </PaperProvider>
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
