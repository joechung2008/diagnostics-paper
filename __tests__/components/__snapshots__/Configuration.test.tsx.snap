// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration renders correctly with config data 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1B1F",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Configuration
  </Text>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        key1
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        value1
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        key2
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        value2
      </Text>
    </View>
    <View
      style={
        {
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        key3
      </Text>
      <Text
        style={
          [
            {
              "flex": 1,
              "minWidth": 0,
              "paddingLeft": 5,
              "paddingRight": 5,
            },
            {
              "color": "#1C1B1F",
              "flex": 1,
              "flexShrink": 1,
              "fontFamily": "System",
            },
          ]
        }
      >
        value3
      </Text>
    </View>
  </View>
</View>
`;

exports[`Configuration renders correctly with empty config 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFBFE",
    }
  }
>
  <Text
    style={
      {
        "color": "#1C1B1F",
        "fontFamily": "System",
        "fontSize": 16,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    Configuration
  </Text>
  <View
    style={
      {
        "backgroundColor": "#FFFBFE",
        "borderColor": "#79747E",
        "borderWidth": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#E7E0EC",
          "borderBottomColor": "#79747E",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Key
      </Text>
      <Text
        style={
          {
            "color": "#1C1B1F",
            "flex": 1,
            "fontFamily": "System",
            "fontWeight": "bold",
            "paddingLeft": 5,
            "paddingRight": 5,
          }
        }
      >
        Value
      </Text>
    </View>
  </View>
</View>
`;
